src/sentry/ingest/userreport.py

```diff
@@ -298,3 +305,44 @@ def _generate_entity_dataset_query(
 def _start_and_end_dates(time: datetime) -> Tuple[datetime, datetime]:
     """Return the 10 min range start and end time range ."""
     return time - timedelta(minutes=5), time + timedelta(minutes=5)
+
+
+def _shim_to_feedback(report, event, project):
+    """
+    takes user reports from the legacy user report endpoint and
+    user reports that come from relay envelope ingestion and
+    creates a new User Feedback from it.
+    User feedbacks are an event type, so we try and grab as much from the
+    legacy user report and event to create the new feedback.
+    """
+    try:
+        feedback_event: dict[str, Any] = {
+            "feedback": {
+                "name": report.get("name", ""),
+                "email": report["email"],
+                "message": report["comments"],
+            },
+            "contexts": {},
+        }
+
+        if event:
```

Comment:
When would we get `event` passed in? Asking because user-feedback can be sent as a stand alone envelope item with only an event_id (no event payload in the same envelope). And the event itself might never have made into to Sentry (or will after the feedback)

Reply:
above in the function, the event is grabbed via the event_id if one was sent. We cover both cases, and will still work if the user report is orphaned.